skypy/halos/_colossus.py

```diff
@@ -72,3 +75,131 @@ def colossus_mass_sampler(redshift, model, mdef, m_min, m_max,
+        sampled over this redshift range.
+    model : string
+        Mass function model which is available in colossus.
+    mdef : str
+        Halo mass definition for spherical overdensities used by colossus.
+    m_min, m_max : float
+        Lower and upper bounds for the halo mass in units of Solar mass, Msun.
+    sky_area : `~astropy.units.Quantity`
+        Sky area over which halos are sampled. Must be in units of solid angle.
+    cosmology : Cosmology
+        Cosmology object to convert comoving density.
+    sigma8 : float
+        Cosmology parameter, amplitude of the (linear) power spectrum on the
+        scale of 8 Mpc/h.
+    ns : float
+        Cosmology parameter, spectral index of scalar perturbation power
+        spectrum.
+    noise : bool, optional
+        Poisson-sample the number of halos. Default is `True`.
+
+    Returns
+    -------
+    redshifts : array_like
+        Redshifts of the halo sample.
+
+    References
+    ----------
+    .. [1] Diemer B., 2018, ApJS, 239, 35
+
+    '''
+    from colossus.cosmology.cosmology import fromAstropy
+    from colossus.lss.mass_function import massFunction
+
+    # Set the cosmology in COLOSSUS
+    fromAstropy(cosmology, sigma8, ns)
+
+    # Integrate the mass function to get the number density of halos at each redshift
+    def dndlnM(lnm, z):
+        return massFunction(np.exp(lnm), z, 'M', 'dndlnM', mdef, model)
+    lnmmin = np.log(m_min/cosmology.h)
+    lnmmax = np.log(m_max/cosmology.h)
+    density = [integrate.quad(dndlnM, lnmmin, lnmmax, args=(z))[0] for z in redshift]
+    density = np.array(density) * np.power(cosmology.h, 3)
+
+    # Sample halo redshifts and assign to bins
+    return redshifts_from_comoving_density(redshift, density, sky_area, cosmology, noise)
+
+
+@units.quantity_input(sky_area=units.sr)
+def colossus_mf(redshift, model, mdef, m_min, m_max, sky_area, cosmology,
+                sigma8, ns, resolution=1000, noise=True):
+    r'''Sample halo redshifts and masses from a COLOSSUS mass function.
+
+    Sample the redshifts and masses of dark matter halos following a mass
+    function implemented in COLOSSUS [1]_ within given mass and redshift ranges
+    and for a given area of the sky.
+
+    Parameters
+    ----------
+    redshift : array_like
+        Defines the edges of a set of redshift bins for which the mass function
+        is evaluated. Must be a monotonically increasing one dimensinoal array
+        of values. Halo redshifts are sampled between the minimum and maximum
+        values in this array.
+    model : string
+        Mass function model which is available in colossus.
+    mdef : str
+        Halo mass definition for spherical overdensities used by colossus.
+    m_min, m_max : float
+        Lower and upper bounds for the halo mass in units of Solar mass, Msun.
+    sky_area : `~astropy.units.Quantity`
+        Sky area over which halos are sampled. Must be in units of solid angle.
```

Comment:
```suggestion
        Sky area over which halos are sampled. Must be in units of solid angle, steradian.
```

Reply:
The units don't have to be steradians, any valid solid angle unit will be handled consistently